src/google/protobuf/descriptor.cc

```diff
@@ -7675,6 +7676,22 @@ bool DescriptorBuilder::OptionInterpreter::ExamineIfOptionIsSet(
   return true;
 }
 
+template <typename T> std::string ValueOutOfRange(
+    absl::string_view type_name, absl::string_view option_name) {
+  return absl::StrFormat(
+    "Value out of range, %d to %d, for %s option \"%s\".", \
+    std::numeric_limits<T>::min(), std::numeric_limits<T>::max(),
+    type_name, option_name);
+}
+
+template <typename T> std::string ValueMustBeInt(
+    absl::string_view type_name, absl::string_view option_name) {
+  return absl::StrFormat(
+    "Value must be integer, from %d to %d, for %s option \"%s\".", \
+    std::numeric_limits<T>::min(), std::numeric_limits<T>::max(),
+    type_name, option_name);
+}
```

Comment:
super nit that I cannot determine from context:  These should be in an anonymous namespace if they are not already

Reply:
Done.

Thanks for the feedback. I'm learning a bit about modern C++ 👍.
I mostly do Go development and haven't done earnest C++ development in like 20 years (which you can probably tell 😆).